tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/rules/MkSeqRule.scala

```diff
@@ -0,0 +1,80 @@
+package at.forsyte.apalache.tla.bmcmt.rules
+
+import at.forsyte.apalache.tla.bmcmt._
+import at.forsyte.apalache.tla.bmcmt.rules.aux.ProtoSeqOps
+import at.forsyte.apalache.tla.bmcmt.types.IntT
+import at.forsyte.apalache.tla.lir.TypedPredefs._
+import at.forsyte.apalache.tla.lir._
+import at.forsyte.apalache.tla.lir.convenience.tla
+import at.forsyte.apalache.tla.lir.oper.ApalacheOper
+import at.forsyte.apalache.tla.lir.storage.BodyMapFactory
+import at.forsyte.apalache.tla.lir.transformations.impl.IdleTracker
+import at.forsyte.apalache.tla.lir.values.TlaInt
+import at.forsyte.apalache.tla.pp.{ConstSimplifier, InlinerOfUserOper, TlaInputError}
+
+/**
+ * Rewriting rule for MkSeq. This rule is similar to [[FoldSeqRule]].
+ *
+ * @author
+ *   Igor Konnov
+ */
+class MkSeqRule(rewriter: SymbStateRewriter) extends RewritingRule {
+  private val proto = new ProtoSeqOps(rewriter)
+  private val simplifier = new ConstSimplifier(new IdleTracker())
+
+  override def isApplicable(symbState: SymbState): Boolean = {
+    symbState.ex match {
+      // match the internal representation of lambda expressions or embedded calls
+      case OperEx(ApalacheOper.mkSeq, _, LetInEx(NameEx(appName), TlaOperDecl(operName, params, _))) =>
+        appName == operName && params.size == 1
+      case _ => false
+    }
+  }
+
+  override def apply(state: SymbState): SymbState = state.ex match {
+    case OperEx(ApalacheOper.mkSeq, lenEx, LetInEx(NameEx(_), opDecl)) =>
+      val capacity = simplifier(lenEx) match {
+        case ValEx(TlaInt(n)) if n.isValidInt && n >= 0 =>
+          n.toInt
+
+        case unexpectedEx =>
+          val msg = "Expected a constant integer expression. Found: " + unexpectedEx
+          throw new TlaInputError(msg, Some(lenEx.ID))
```

Comment:
This error might be confusing, because it collapses 3 cases:
-  if you get a number that fits a `BigInt`, but not an `Int`. 
- if you get a negative number
- if you don't get a `TlaInt` at all

I'd suggest making an explicit error message for "number too large" and "number negative".